certora/spec/timelockAuthorizerTed.spec

```diff
@@ -37,11 +38,14 @@ rule scheduledExecutionCanBeCancelledOnlyOnce(env e, uint256 index) {
     bool canceled_before = getSchedExeCancelled(index);
     bool executed_before = getSchedExeExecuted(index);
 
-    // require(canceled_before && !executed_before || !canceled_before && executed_before);
+    // require(canceled_before && !executed_before || !canceled_before && executed_before); <== needed?
     require(canceled_before || executed_before);
 
     cancel@withrevert(e, index);
     assert lastReverted;
+    assert ( canceled_before || executed_before ) => lastReverted; 
+
+    // what's the difference? when to use what ?
```

Comment:
Let me compare the two following options:

1. Requirement `require(canceled_before || executed_before);` and assertion `assert lastReverted;`
2. Implication `assert ( canceled_before || executed_before ) => lastReverted; `

The difference is that when choosing 1. we narrow the problem and make it easier for the SMT team to optimise: I have spoken to Jaroslav and he says that 1. is better than 2. and that whenever we want to require something, it is better to do it as soon as possible and when we choose between such implication `assert A => B` and we are not interested in knowing what would happen when `A` is not true, it is better to `require A` and then just `assert B`.